engine/script/src/test/test_json.lua

```diff
@@ -41,28 +41,28 @@ function assert_error(js, expected)
 end
 
 function test_json_invalid_primitive()
-    assert_error("Null", "Invalid JSON primitive: Null")
-    assert_error("NULL", "Invalid JSON primitive: NULL")
-    assert_error("True", "Invalid JSON primitive: True")
-    assert_error("TRUE", "Invalid JSON primitive: TRUE")
-    assert_error("False", "Invalid JSON primitive: False")
-    assert_error("FALSE", "Invalid JSON primitive: FALSE")
-    assert_error("defold", "Invalid JSON primitive: defold")
-    assert_error("0.d3", "Invalid JSON primitive: 0.d3")
-    assert_error("{1 2 3}", "Incomplete JSON object: {1 2 3}")
-    assert_error("{1: 2, 3}", "Incomplete JSON object: {1: 2, 3}")
-    assert_error("{ response = \"ok\" }", "Incomplete JSON object: { response = \"ok\" }")
+    print("Expected parsing errors ->")
+    assert_error("Null", "Expected value but found invalid token at character 1")
+    assert_error("NULL", "Expected value but found invalid token at character 1")
+    assert_error("True", "Expected value but found invalid token at character 1")
+    assert_error("TRUE", "Expected value but found invalid token at character 1")
+    assert_error("False", "Expected value but found invalid token at character 1")
+    assert_error("FALSE", "Expected value but found invalid token at character 1")
+    assert_error("defold", "Expected value but found invalid token at character 1")
+    assert_error("0.d3", "Expected the end but found invalid token at character 3")
+    assert_error("{1 2 3}", "Expected object key string but found T_NUMBER at character 2")
+    assert_error("{1: 2, 3}", "Expected object key string but found T_NUMBER at character 2")
+    assert_error("{ response = \"ok\" }", "Expected object key string but found invalid token at character 3")
+    print("<- end of expected parsing errors.")
```

Comment:
Not entirely sure what to do here, since I think the parsing errors are a bit lacking.
However, this is still what users have used for the defold-cjson extension.